src/App.tsx

```diff
@@ -1,61 +1,148 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
-import classNames from 'classnames';
+import cn from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getAllUsers } from './api/users';
+import { Post } from './types/Post';
+import { getPosts } from './api/posts';
+import { addComment, deleteCommentById, getCommentById } from './api/comments';
+import { Comment } from './types/Comment';
+
+const ERROR = 'Something went wrong!';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [selectUser, setSelectUser] = useState<User | null>(null);
+  const [selectPost, setSelectPost] = useState<Post | null>(null);
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getAllUsers()
+      .then(setUsers)
+      .catch(() => setError(ERROR))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (selectUser !== null) {
```

Comment:
Fix such conditions everywhere
```suggestion
    if (selectUser) {
```